numba/core/untyped_passes.py

```diff
@@ -910,6 +910,17 @@ def inject_loop_body(self, switch_ir, loop_ir, caller_max_label,
                 new_var_dict[name] = mk_unique_var(name)
             replace_var_names(loop_blocks, new_var_dict)
 
+            # Patch up the scope with the new variable names, this is horrible
+            new_var_table = get_name_var_table(loop_blocks)
+            for blk in loop_blocks.values():
+                scope = blk.scope
+                real_con = scope.localvars._con
+                for name, var in dict(scope.localvars._con).items():
+                    if name in new_var_dict:
+                        new_name = new_var_dict[name]
+                        del real_con[name]
+                        real_con[new_name] = new_var_table[new_name]
```

Comment:
I probably will make a follow up PR to refactor it with `ir_utils.fixup_var_define_in_scope`

Reply:
Thanks. I think the issue is that the versioned loop bodied have versions of _some_ of the original variables and potentially replace the original.